timescaledb/tutorials/ingest-real-time-websocket-data.md

```diff
@@ -0,0 +1,347 @@
+# Ingest real-time financial websocket data
+
+This tutorial shows you how to ingest real-time time-series data into
+TimescaleDB using a websocket connection. The tutorial sets up a data pipeline to
+ingest real-time data from our data partner, [Twelve Data][twelve-data].
+Twelve Data provides a number of different financial APIs, including stock,
+crypto, forex, ETFs, etc... It also supports websocket connections in case
+you want to update your database frequently. With websockets, you need to
+connect to the server, subscribe to symbols, and you can start receiving data
+in real-time during market hours.
+
+Once you complete this tutorial, you'll have a data pipeline set
+up that ingests real-time financial data into your TimescaleDB instance.
+
+This tutorial uses Python and the
+API [wrapper library][twelve-wrapper] provided
+by Twelve Data.
+
+## Prerequisites
+- A TimescaleDB instance running locally or on the cloud. For more information,
+[see installation options][install-ts]
+- Python 3
+- Sign up for [Twelve Data][twelve-signup] (there's a free tier!)
+
+## Set up a new Python environment
+
+Create a new Python virtual environment for this project and activate it. All
+the packages you need to complete for this tutorial will be installed in this environment.
+
+<procedure>
+
+### Setting up a new Python environment
+
+1. Create and activate a Python virtual environment
+    ```bash
+    virtualenv env
+    source env/bin/activate
+    ``` 
+1. Install the Twelve Data Python [wrapper library](https://github.com/twelvedata/twelvedata-python) with websocket support:
+    ```bash
+    pip install twelvedata websocket-client
+    ```
+    This library makes it easy to make requests to the API and maintain a stable websocket connection.
```

Comment:
Move this sentence above the code block.

Reply:
https://github.com/timescale/docs/commit/deebb6d576eb352c8238cf8f346a97a6a14b9209